pkgs/os-specific/darwin/libcxx/default.nix

```
@@ -0,0 +1,26 @@
+{
+  apple-sdk,
+  stdenvNoCC,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "libcxx";
+  inherit (apple-sdk) src version;
```

I’m a bit worried that macOS SDK versions look enough like LLVM versions that people might be smart and try to condition on `libc++` version and fail. But I’m not sure if we can do anything better, either…

---

I don’t know what to put. There’s enough misunderstanding around what version of Clang is shipped by Apple that people might reflexively go with the LLVM one because it is slightly newer, or they might be confused when 14.x has an actually older runtime (or 26.0 has a newer one).

Are there any examples of conditioning on libc++ version?

---

`overrideLibcxx` in `pkgs/stdenv/adapters.nix` does, I wonder who we should blame for that? 😆

But since Swift and Qt 5 are the only user (and hopefully Swift won’t be soon? and the Qt 5 one seems dubious to me), that’s hopefully not the end of the world; we can potentially just drop it and the assertion is harmless anyway since 15 > 12.

This is not a blocker for me, but it would be nice for the package name in the store path to be distinguished from an LLVM one, and maybe we can solve both problems at once. Some proposals:

* `libcxx-apple-sdk-15.2`
* `libcxx-0-apple-sdk-15.2` (ensures version comparisons will at least fail harder)

I think that *ideally*, we’d have something like `libcxx-⟨closest upstream LLVM version⟩-apple-sdk-⟨macOS SDK version⟩`. But it might be hard to reliably determine the corresponding LLVM version, given what Apple’s tree is like.